integration/docker/run_test.go

```diff
@@ -112,16 +110,23 @@ func createLoopDevice() (string, string, error) {
 	}
 
 	// create loop device
-	for i := 0; i < losetupMaxTries; i++ {
-		loopPath := fmt.Sprintf("/dev/loop%d", i)
-		losetupCmd := NewCommand("losetup", "-P", loopPath, f.Name())
-		_, _, exitCode := losetupCmd.Run()
-		if exitCode == 0 {
-			return f.Name(), loopPath, nil
-		}
+	losetupCmd := NewCommand("losetup", "-fP", f.Name())
+	if _, stderr, exitCode := losetupCmd.Run(); exitCode != 0 {
+		return "", "", fmt.Errorf("%s", stderr)
 	}
 
-	return "", "", fmt.Errorf("unable to create loop device for disk %s", f.Name())
+	// get loop device path
+	getLoopPath := NewCommand("losetup", "-j", f.Name())
+	stdout, stderr, exitCode := getLoopPath.Run()
+	if exitCode != 0 {
+		return "", "", fmt.Errorf("%s", stderr)
+	}
+	re := regexp.MustCompile("/dev/loop[0-9]+")
+	loopPath := re.FindStringSubmatch(stdout)
+	if len(loopPath) == 0 {
+		return "", "", fmt.Errorf("Unable to get loop device path")
```

Comment:
Nit: I'd consider adding `stdout` and `stderr` [*] to the returned error to make it easier to see what went wrong.

---

[*] - which we'd *hope* was empty if `exitCode == 0`, but who knows? :)